src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -782,4 +792,20 @@ public boolean isCourseArchived(CourseAttributes course, InstructorAttributes in
         }
         return result;
     }
+    
+    public List<String> getArchivedCourseIds(List<CourseDetailsBundle> allCourses, List<InstructorAttributes> instructorList) {
+        List<String> archivedCourseIds = new ArrayList<String>();
+        for (CourseDetailsBundle cdb : allCourses) {
+            if (cdb.course.isArchived) {
+                archivedCourseIds.add(cdb.course.id);
+            }
+        }
+        for (InstructorAttributes instructor : instructorList) {
+            if (instructor.isArchived != null && instructor.isArchived == true) {
+                archivedCourseIds.add(instructor.courseId);
```

Comment:
Actually the logic here does not really make sense. If I recall, instructor's isArchived should take precedence over the course's isArchived attribute. If a course isArchived is true, but the instructor's isArchived is false, then the course should not be archived for the instructor.